src/index.ts

```diff
@@ -0,0 +1,326 @@
+      return [
+        {
+          kind: 'LABEL',
+          text: 'No blocks have been used yet!',
+        },
+      ];
+    }
+
+    return blockTypeList.slice(0, this.numBlocksPerCategory).map((key) => {
+      const json = this.defaultJsonForBlockLookup[key] || {};
+      return {
+        ...json,
+        kind: 'BLOCK',
+        type: key,
+        x: undefined,
+        y: undefined,
+      };
+    });
+  };
+
+  /**
+   * Event listener for workspace events.
+   *
+   * @param e - The event object for workspace events
+   */
+  eventListener(e: Blockly.Events.Abstract): void {
+    if (e.type === Blockly.Events.FINISHED_LOADING) {
+      this.workspaceHasFinishedLoading = true;
+      return;
+    }
+
+    if (
+      e.type === Blockly.Events.BLOCK_CREATE &&
+      this.workspaceHasFinishedLoading &&
+      (e as Blockly.Events.BlockCreate).json?.type
+    ) {
+      const newBlockType = (e as Blockly.Events.BlockCreate).json.type;
+
+      // If this is the first time creating this block, store its default
+      // configuration so we know how exactly to render it in the toolbox.
+      if (!this.defaultJsonForBlockLookup[newBlockType]) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        this.defaultJsonForBlockLookup[newBlockType] = (e as any).json;
+      }
+      this.recentlyUsedBlocks.unshift(newBlockType);
+    }
+  }
+  /**
+   * Saves the state of this object to a serialized JSON.
+   *
+   * @returns a serialized data object including this object's state
+   */
+  saveToSerializedData(): object {
+    return {
+      recentlyUsedBlocks: this.recentlyUsedBlocks,
+      defaultJsonForBlockLookup: this.defaultJsonForBlockLookup,
+    };
+  }
+
+  /**
+   * Loads the state of this object from a serialized JSON.
+   *
+   * @param state the serialized data payload to load from
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  loadFromSerializedData(state: any): void {
+    if (state.recentlyUsedBlocks) {
+      this.recentlyUsedBlocks = state.recentlyUsedBlocks;
+    }
+    if (state.defaultJsonForBlockLookup) {
+      this.defaultJsonForBlockLookup = state.defaultJsonForBlockLookup;
+    }
+  }
+
+  /**
+   * Resets the internal state of this object.
+   */
+  clearPriorBlockData(): void {
+    this.recentlyUsedBlocks = [];
+    this.defaultJsonForBlockLookup = {};
+  }
+}
+
+/**
+ * Main entry point to initialize the suggested blocks categories.
+ *
+ * @param workspace the workspace to load into
+ * @param numBlocksPerCategory how many blocks should be included per
+ * category. Defaults to 10.
+ * @param waitForFinishedLoading whether to wait until we hear the
+ * FINISHED_LOADING event before responding to BLOCK_CREATE events. Set to false
+ * if you disable events during initial load. Defaults to true.
+ */
+export const init = function (
+  workspace: Blockly.WorkspaceSvg,
+  numBlocksPerCategory: 10,
+  waitForFinishedLoading: true,
```

Comment:
Need to change to `= 10` and `= true` as it was originally to provide default values if these inputs weren't supplied. The way this is expressed, you're saying the inputs have to be `10` and `true`, making them useless!